packages/discovery-react-components/src/components/DocumentPreview/utils/documentData.ts

```diff
@@ -24,3 +30,15 @@ export function getTextMappings(
   }
   return mappings;
 }
+
+/**
+ * Get `file_type` document property as a string. Usually, this
+ * prop is stored as a JSON string.
+ */
+export function getDocumentType(doc: QueryResult | null | undefined): string | null {
+  let docType = get(doc, 'extracted_metadata.file_type');
+  if (docType && typeof docType === 'string') {
+    return docType;
+  }
+  return null;
+}
```

Comment:
I think it would make the code cleaner if we simply had `isJsonFile(doc)` and `isCsvFile(doc)` functions, which return a boolean. That way, you won't have to export both a function and constants, only to do `result === CONST` -- just handle all those details here.